# Change Log

## [Unreleased](https://github.com/sanger-pathogens/gff3toembl/tree/HEAD)

[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.1.4...HEAD)

**Closed issues:**

- Installation and running problems [\#74](https://github.com/sanger-pathogens/gff3toembl/issues/74)
- mac brew install does not work. [\#63](https://github.com/sanger-pathogens/gff3toembl/issues/63)

**Merged pull requests:**

- fix readme and add to codecov [\#75](https://github.com/sanger-pathogens/gff3toembl/pull/75) ([ssjunnebo](https://github.com/ssjunnebo))
- using six for urllib.parse for support of python 3 [\#73](https://github.com/sanger-pathogens/gff3toembl/pull/73) ([sachalau](https://github.com/sachalau))
- update LICENSE and AUTHORS [\#72](https://github.com/sanger-pathogens/gff3toembl/pull/72) ([ssjunnebo](https://github.com/ssjunnebo))

## [v1.1.4](https://github.com/sanger-pathogens/gff3toembl/tree/v1.1.4) (2016-11-11)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.1.3...v1.1.4)

**Merged pull requests:**

- Include XX line between features \(FT\) and sequence \(SQ\) [\#70](https://github.com/sanger-pathogens/gff3toembl/pull/70) ([peterjc](https://github.com/peterjc))
- Unescape percent encoding from GFF3, then line wrap [\#69](https://github.com/sanger-pathogens/gff3toembl/pull/69) ([peterjc](https://github.com/peterjc))

## [v1.1.3](https://github.com/sanger-pathogens/gff3toembl/tree/v1.1.3) (2016-11-10)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.1.2...v1.1.3)

**Merged pull requests:**

- More wrapping tests; Fix ignored test [\#68](https://github.com/sanger-pathogens/gff3toembl/pull/68) ([peterjc](https://github.com/peterjc))

## [v1.1.2](https://github.com/sanger-pathogens/gff3toembl/tree/v1.1.2) (2016-11-10)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.1.1...v1.1.2)

**Closed issues:**

- Sequence header SQ line can exceed 80 characters [\#65](https://github.com/sanger-pathogens/gff3toembl/issues/65)

**Merged pull requests:**

- EMBL files wrap at 80 characters, not 79 [\#67](https://github.com/sanger-pathogens/gff3toembl/pull/67) ([peterjc](https://github.com/peterjc))

## [v1.1.1](https://github.com/sanger-pathogens/gff3toembl/tree/v1.1.1) (2016-11-09)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.1.0...v1.1.1)

**Closed issues:**

- Change PROKKA to Prokka in repo description [\#64](https://github.com/sanger-pathogens/gff3toembl/issues/64)
- linuxbrew recipe needs updating [\#62](https://github.com/sanger-pathogens/gff3toembl/issues/62)

**Merged pull requests:**

- Sq exceeds 80 [\#66](https://github.com/sanger-pathogens/gff3toembl/pull/66) ([andrewjpage](https://github.com/andrewjpage))

## [v1.1.0](https://github.com/sanger-pathogens/gff3toembl/tree/v1.1.0) (2016-09-21)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.9...v1.1.0)

**Implemented enhancements:**

- I have added this to my brew repo \(updated to 1.0.9\) [\#59](https://github.com/sanger-pathogens/gff3toembl/issues/59)

**Fixed bugs:**

- Failure to install on Biolinux [\#38](https://github.com/sanger-pathogens/gff3toembl/issues/38)

**Merged pull requests:**

- fix typos [\#61](https://github.com/sanger-pathogens/gff3toembl/pull/61) ([satta](https://github.com/satta))
- Docker file [\#60](https://github.com/sanger-pathogens/gff3toembl/pull/60) ([andrewjpage](https://github.com/andrewjpage))

## [v1.0.9](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.9) (2016-05-24)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.8...v1.0.9)

**Merged pull requests:**

- check codon start is a number [\#57](https://github.com/sanger-pathogens/gff3toembl/pull/57) ([andrewjpage](https://github.com/andrewjpage))

## [v1.0.8](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.8) (2016-05-23)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.6...v1.0.8)

**Merged pull requests:**

- Remove colour tag [\#56](https://github.com/sanger-pathogens/gff3toembl/pull/56) ([andrewjpage](https://github.com/andrewjpage))
- Remove double quoting [\#55](https://github.com/sanger-pathogens/gff3toembl/pull/55) ([andrewjpage](https://github.com/andrewjpage))
- Break on hyphen [\#54](https://github.com/sanger-pathogens/gff3toembl/pull/54) ([andrewjpage](https://github.com/andrewjpage))

## [v1.0.6](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.6) (2016-05-06)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.5...v1.0.6)

## [v1.0.5](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.5) (2016-05-03)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.3...v1.0.5)

**Implemented enhancements:**

- Format for listing authors not defined [\#45](https://github.com/sanger-pathogens/gff3toembl/issues/45)

**Fixed bugs:**

- Should unescape %2C in GFF column 9 as comma in EMBL [\#50](https://github.com/sanger-pathogens/gff3toembl/issues/50)
- Document versions of Python supported [\#42](https://github.com/sanger-pathogens/gff3toembl/issues/42)

**Merged pull requests:**

- Avoid breaking long words in product attributes [\#53](https://github.com/sanger-pathogens/gff3toembl/pull/53) ([peterjc](https://github.com/peterjc))
- Un-escape %2C to comma etc for GFF3 to EMBL [\#52](https://github.com/sanger-pathogens/gff3toembl/pull/52) ([peterjc](https://github.com/peterjc))
- Ec dash [\#49](https://github.com/sanger-pathogens/gff3toembl/pull/49) ([andrewjpage](https://github.com/andrewjpage))

## [v1.0.3](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.3) (2015-12-10)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.2...v1.0.3)

**Fixed bugs:**

- Long author lists break instead of line wrapping [\#46](https://github.com/sanger-pathogens/gff3toembl/issues/46)

**Merged pull requests:**

- Make publication arguments a little more explicit [\#48](https://github.com/sanger-pathogens/gff3toembl/pull/48) ([peterjc](https://github.com/peterjc))

## [v1.0.2](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.2) (2015-12-10)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.1...v1.0.2)

**Closed issues:**

- self.assertItemsEqual is Python 2.7 only [\#43](https://github.com/sanger-pathogens/gff3toembl/issues/43)
- Non-standard indentation of Python code [\#41](https://github.com/sanger-pathogens/gff3toembl/issues/41)
- Python 3 support [\#40](https://github.com/sanger-pathogens/gff3toembl/issues/40)

**Merged pull requests:**

- Wrap publication lines [\#47](https://github.com/sanger-pathogens/gff3toembl/pull/47) ([andrewjpage](https://github.com/andrewjpage))
- Add note that it only works on Python 2.7 [\#44](https://github.com/sanger-pathogens/gff3toembl/pull/44) ([bewt85](https://github.com/bewt85))
- Use Python 3 ready exception, print, etc [\#39](https://github.com/sanger-pathogens/gff3toembl/pull/39) ([peterjc](https://github.com/peterjc))

## [v1.0.1](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.1) (2015-09-02)
[Full Changelog](https://github.com/sanger-pathogens/gff3toembl/compare/v1.0.0...v1.0.1)

**Closed issues:**

- v1.0.0 vs v1.0 [\#36](https://github.com/sanger-pathogens/gff3toembl/issues/36)

**Merged pull requests:**

- Complete genome naming [\#37](https://github.com/sanger-pathogens/gff3toembl/pull/37) ([andrewjpage](https://github.com/andrewjpage))
- README.md: Fix a spelling mistake gff3toemble [\#35](https://github.com/sanger-pathogens/gff3toembl/pull/35) ([sjackman](https://github.com/sjackman))

## [v1.0.0](https://github.com/sanger-pathogens/gff3toembl/tree/v1.0.0) (2015-07-13)
**Closed issues:**

- OSX and dlylib [\#33](https://github.com/sanger-pathogens/gff3toembl/issues/33)
- Does this cope with Prokka GFF3 files ok? [\#23](https://github.com/sanger-pathogens/gff3toembl/issues/23)

**Merged pull requests:**

- Add TravisCI Support \(and fix setup.py\) [\#34](https://github.com/sanger-pathogens/gff3toembl/pull/34) ([bewt85](https://github.com/bewt85))
- case insensitive hypo search [\#32](https://github.com/sanger-pathogens/gff3toembl/pull/32) ([andrewjpage](https://github.com/andrewjpage))
- get the version of the script with --version [\#31](https://github.com/sanger-pathogens/gff3toembl/pull/31) ([andrewjpage](https://github.com/andrewjpage))
- Empty product hypo protein [\#30](https://github.com/sanger-pathogens/gff3toembl/pull/30) ([andrewjpage](https://github.com/andrewjpage))
- Update README and version [\#29](https://github.com/sanger-pathogens/gff3toembl/pull/29) ([bewt85](https://github.com/bewt85))
- gff3toembl on Mac OS X [\#28](https://github.com/sanger-pathogens/gff3toembl/pull/28) ([satta](https://github.com/satta))
- Add usage example to readme [\#26](https://github.com/sanger-pathogens/gff3toembl/pull/26) ([bewt85](https://github.com/bewt85))
- Default "publication" to include today's date [\#25](https://github.com/sanger-pathogens/gff3toembl/pull/25) ([bewt85](https://github.com/bewt85))
- Refactor / rewrite / remove convert.py [\#24](https://github.com/sanger-pathogens/gff3toembl/pull/24) ([bewt85](https://github.com/bewt85))
- Duplicate entries [\#22](https://github.com/sanger-pathogens/gff3toembl/pull/22) ([andrewjpage](https://github.com/andrewjpage))
- intermediate gff prefix to suffic [\#21](https://github.com/sanger-pathogens/gff3toembl/pull/21) ([andrewjpage](https://github.com/andrewjpage))
- sort and tidy gff files with GT [\#20](https://github.com/sanger-pathogens/gff3toembl/pull/20) ([andrewjpage](https://github.com/andrewjpage))
- Revert rna feature translate [\#19](https://github.com/sanger-pathogens/gff3toembl/pull/19) ([andrewjpage](https://github.com/andrewjpage))
- No trans table for rna [\#18](https://github.com/sanger-pathogens/gff3toembl/pull/18) ([andrewjpage](https://github.com/andrewjpage))
- Convert type for 16s [\#17](https://github.com/sanger-pathogens/gff3toembl/pull/17) ([andrewjpage](https://github.com/andrewjpage))
- Unknown function [\#16](https://github.com/sanger-pathogens/gff3toembl/pull/16) ([andrewjpage](https://github.com/andrewjpage))
- replace unknown with uncharacterised [\#15](https://github.com/sanger-pathogens/gff3toembl/pull/15) ([andrewjpage](https://github.com/andrewjpage))
- extra dot [\#14](https://github.com/sanger-pathogens/gff3toembl/pull/14) ([andrewjpage](https://github.com/andrewjpage))
- passes java validator [\#13](https://github.com/sanger-pathogens/gff3toembl/pull/13) ([andrewjpage](https://github.com/andrewjpage))
- Update format [\#12](https://github.com/sanger-pathogens/gff3toembl/pull/12) ([andrewjpage](https://github.com/andrewjpage))
- version [\#11](https://github.com/sanger-pathogens/gff3toembl/pull/11) ([andrewjpage](https://github.com/andrewjpage))
- Made EC number values unique [\#10](https://github.com/sanger-pathogens/gff3toembl/pull/10) ([aslett1](https://github.com/aslett1))
- Create Chromosome.txt file [\#9](https://github.com/sanger-pathogens/gff3toembl/pull/9) ([andrewjpage](https://github.com/andrewjpage))
- update version [\#8](https://github.com/sanger-pathogens/gff3toembl/pull/8) ([andrewjpage](https://github.com/andrewjpage))
- Add chrom name to source [\#7](https://github.com/sanger-pathogens/gff3toembl/pull/7) ([andrewjpage](https://github.com/andrewjpage))
- Split lines on words, fix headers and references [\#6](https://github.com/sanger-pathogens/gff3toembl/pull/6) ([andrewjpage](https://github.com/andrewjpage))
- fix input param names [\#5](https://github.com/sanger-pathogens/gff3toembl/pull/5) ([andrewjpage](https://github.com/andrewjpage))
- convert inference to db\_xref [\#4](https://github.com/sanger-pathogens/gff3toembl/pull/4) ([andrewjpage](https://github.com/andrewjpage))
- rename script [\#3](https://github.com/sanger-pathogens/gff3toembl/pull/3) ([andrewjpage](https://github.com/andrewjpage))
- Working script for converting [\#2](https://github.com/sanger-pathogens/gff3toembl/pull/2) ([andrewjpage](https://github.com/andrewjpage))
- initial working version [\#1](https://github.com/sanger-pathogens/gff3toembl/pull/1) ([andrewjpage](https://github.com/andrewjpage))

